onnxruntime/contrib_ops/webgpu/bert/flash_attention.cc

```diff
@@ -275,8 +275,23 @@ Status FlashAttentionProgram::GenerateShaderCode(ShaderHelper& shader) const {
 
   var previous_max : q_element_t = min_value;
   var previous_denom : q_element_t = 0;
+)MAIN_FN";
 
-  for(var k_start = 0u; k_start < uniforms.total_sequence_length; k_start+=capped_sg_size)
+  if (is_unidirectional_) {
```

Comment:
Why you can't keep to use `uniforms.is_gqa` but use `is_unidirectional_`?

Reply:
The only purpose of `is_gqa` in the shader is to check if the attention mechanism is unidirectional. 
Renaming it to `is_unidirectional` would be more accurate and self-documenting.

The current name also implies that:
- Grouped-Query Attention (GQA) is always unidirectional 
- and Multi-Head Attention (MHA) is always bidirectional. 

The assumption would change, that we may want to apply a unidirectional to MHA in the future.